### ISSUE TITLE
Branch not being removed

### ISSUE THREAD

**Issue by concertcoder** [GitbookIO/gitbook#1679]
### What is the current behavior?
There's an extra branch called 'revert-3-feat-redux' showing on our gitbook (https://angular-2-training-book.rangle.io/) that does not actually exist on the github repo.  We've deleted it.  I tried pushing up a branch of the same name (that's giving us issues) with another branch, seeing both branches, removing them, but only the other branch was removed.  It seems like we're stuck with this extra branch.

TL;DR There's an extra branch we've deleted that is not being removed from the gitbook data store.

### What is the expected behavior?
Branches should match branches on github repo.

Any idea how to get rid of it even if it's a round about way?
We tested calling the API for gitbook directly and it lives there, so this is a syncing issue of some sort we think.

@bennett000 this is the open ticket

**Comment by Mackiovello** [comment]
I'm having this issue too. Very annoying when you're using it with the `versions` plugin

**Comment by bennett000** [comment]
@concertcoder thanks for making this issue.  I did a search and it [looks like GitBook has a formal reporting process](https://www.gitbook.com/contact "Reporting GitBook Issues").

Based on the open issues here, the outstanding PR's, and the steady (albeit irregular) commits, it seems like they probably use other tools to manage their stories.  

I filed an incident report there.  @Mackiovello you might want to let them know about your specific case.

**Comment by AaronO** [comment]
@concertcoder @Mackiovello @bennett000 Thanks for reporting this.

This is a result of a change we deployed in our GitHub syncing strategy. We changed how our Git Server syncs with GitHub to improve robustness and in a few days we'll be shipping our new GitHub integration that uses GitHub's new "integration API". The goals of those two changes are to reduce errors people would encounter when editing from GitBook.com, especially in teams (e.g: when someone is a gitbook collaborator but doesn't have a GitHub account)

I'll look into fixing this soon enough, expect a fix deployed before mid of next week. I think it comes down to the simple fact that we currently don't run our sync logic when we receive "delete branch" webhooks from GitHub, we only process "push" events.

**Comment by bennett000** [comment]
@AaronO thanks so much for the quick response and the action plan!
